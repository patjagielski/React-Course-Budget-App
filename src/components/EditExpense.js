import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startEditExpense, startRemoveExpense} from '../actions/expenses';


export class EditExpensePage extends React.Component{
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    render(){
        return(
            <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Edit Expense</h1>
                </div>
            </div>
            <div className="content-container">
                <ExpenseForm
                    expense={ this.props.expense } 
                    onSubmit={ this.onSubmit }
                />
                <button className="button-layout__remove" onClick={this.onRemove}>Remove</button>
            </div>
        </div>
        );
    }
};


const mapSateToProps = (state, props) => ({
    expense: state.expenses.find((expense)=> expense.id === props.match.params.id)

});

const mapDispatchToProps = (dispatch, props) => ({
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense))
});

export default connect(mapSateToProps, mapDispatchToProps)(EditExpensePage);