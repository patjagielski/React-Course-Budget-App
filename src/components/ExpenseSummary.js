import React from 'react';
import numeral from 'numeral';
import {connect} from 'react-redux';
import selectTotalExpense from '../selectors/expenses-total';
import selectExpenses from'../selectors/expenses';

export const ExpenseSummary = ({expenseCount, expenseTotal}) => {
    const expenseWord = expenseCount > 1 ? 'expenses' : "expense";
    const formatExpenseTotal = numeral(expenseTotal / 100).format('$0,0.00');
    return (
    <div>
       <h1>Viewing {expenseCount} {expenseWord} totalling {formatExpenseTotal}</h1>
    </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
      expenseCount: visibleExpenses.length,
      expenseTotal: selectTotalExpense(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);