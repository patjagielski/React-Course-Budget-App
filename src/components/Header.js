import React from 'react';
import ReactDOM from 'react-dom';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Budget Application</h1>
        <NavLink to='/' activeClassName='is-active' exact={true}>Home</NavLink>
        <br />
        <NavLink to='/create' activeClassName='is-active'>Create Expense</NavLink>
        <br />
    </header>
);

export default Header;



// <NavLink to='/help' activeClassName='is-active'>Help Page</NavLink>