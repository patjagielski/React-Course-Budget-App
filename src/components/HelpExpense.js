import React from 'react';
import ReactDOM from 'react-dom';

const HelpExpensePage = () => (
    <div>
        This is the Help Component
    </div>
);

export default HelpExpensePage;