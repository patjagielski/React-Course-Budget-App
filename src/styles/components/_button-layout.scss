.button-layout{
    background: $blue;
    border: none;
    color: white;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link{
 background: none;   
}

.button-layout__remove{
    @extend .button-layout;
    background: rgb(134, 33, 33);
}