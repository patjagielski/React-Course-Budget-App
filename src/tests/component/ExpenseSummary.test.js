import {ExpenseSummary} from '../../components/ExpenseSummary';
import React from 'react';
import { shallow } from'enzyme';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount = {1} expenseTotal={240}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount = {12} expenseTotal={24012312}/>);
    expect(wrapper).toMatchSnapshot();
});