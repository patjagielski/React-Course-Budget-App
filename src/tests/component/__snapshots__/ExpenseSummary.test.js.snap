// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.40
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    12
     
    expenses
     totalling 
    $240,123.12
  </h1>
</div>
`;
